### ISSUE TITLE
Running Vim in the terminal results in nonsense "settle:settle:" text

### ISSUE THREAD

**Issue by lfairy** [Microsoft/vscode#33173]
- VSCode Version: Code 1.15.1 (41abd21afdf7424c89319ee7cb0445cc6f376959, 2017-08-16T17:16:27.618Z) installed via AUR
- OS Version: Linux x64 4.12.8-2-ARCH
- Extensions: 

Extension|Author (truncated)|Version
---|---|---
tslint|eg2|0.17.0

---

Steps to Reproduce:

1. In the terminal, execute `vim`.
2. Observe the text `settle:settle:` in the buffer.

This text can be edited as usual. Saving the file will result in `settle:settle:` being written to disk. Opening Vim to an existing file will prepend `settle:settle:` to the contents of that file.

![screenshot from 2017-08-26 13-48-16](https://user-images.githubusercontent.com/1572323/29737630-3bf0b00e-8a65-11e7-8fe4-286f20dd2db1.png)

Reproduces without extensions: Yes

**Comment by lfairy** [comment]
```
~ ) vim --version
VIM - Vi IMproved 8.0 (2016 Sep 12, compiled Aug 23 2017 23:22:13)
Included patches: 1-987
Compiled by Arch Linux
Huge version with GTK3 GUI.  Features included (+) or not (-):
+acl             +file_in_path    +mouse_sgr       +tag_old_static
+arabic          +find_in_path    -mouse_sysmouse  -tag_any_white
+autocmd         +float           +mouse_urxvt     +tcl/dyn
+balloon_eval    +folding         +mouse_xterm     +termguicolors
+browse          -footer          +multi_byte      -terminal
++builtin_terms  +fork()          +multi_lang      +terminfo
+byte_offset     +gettext         -mzscheme        +termresponse
+channel         -hangul_input    +netbeans_intg   +textobjects
+cindent         +iconv           +num64           +timers
+clientserver    +insert_expand   +packages        +title
+clipboard       +job             +path_extra      +toolbar
+cmdline_compl   +jumplist        +perl/dyn        +user_commands
+cmdline_hist    +keymap          +persistent_undo +vertsplit
+cmdline_info    +lambda          +postscript      +virtualedit
+comments        +langmap         +printer         +visual
+conceal         +libcall         +profile         +visualextra
+cryptv          +linebreak       +python/dyn      +viminfo
+cscope          +lispindent      +python3/dyn     +vreplace
+cursorbind      +listcmds        +quickfix        +wildignore
+cursorshape     +localmap        +reltime         +wildmenu
+dialog_con_gui  +lua/dyn         +rightleft       +windows
+diff            +menu            +ruby/dyn        +writebackup
+digraphs        +mksession       +scrollbind      +X11
+dnd             +modify_fname    +signs           -xfontset
-ebcdic          +mouse           +smartindent     +xim
+emacs_tags      +mouseshape      +startuptime     -xpm
+eval            +mouse_dec       +statusline      +xsmp_interact
+ex_extra        +mouse_gpm       -sun_workshop    +xterm_clipboard
+extra_search    -mouse_jsbterm   +syntax          -xterm_save
+farsi           +mouse_netterm   +tag_binary      
   system vimrc file: "/etc/vimrc"
     user vimrc file: "$HOME/.vimrc"
 2nd user vimrc file: "~/.vim/vimrc"
      user exrc file: "$HOME/.exrc"
  system gvimrc file: "/etc/gvimrc"
    user gvimrc file: "$HOME/.gvimrc"
2nd user gvimrc file: "~/.vim/gvimrc"
       defaults file: "$VIMRUNTIME/defaults.vim"
    system menu file: "$VIMRUNTIME/menu.vim"
  fall-back for $VIM: "/usr/share/vim"
Compilation: gcc -c -I. -Iproto -DHAVE_CONFIG_H -DFEAT_GUI_GTK  -pthread -I/usr/include/gtk-3.0 -I/usr/include/at-spi2-atk/2.0 -I/usr/include/at-spi-2.0 -I/usr/include/dbus-1.0 -I/usr/lib/dbus-1.0/include -I/usr/include/gtk-3.0 -I/usr/include/gio-unix-2.0/ -I/usr/include/cairo -I/usr/include/pango-1.0 -I/usr/include/atk-1.0 -I/usr/include/cairo -I/usr/include/pixman-1 -I/usr/include/freetype2 -I/usr/include/libpng16 -I/usr/include/harfbuzz -I/usr/include/glib-2.0 -I/usr/lib/glib-2.0/include -I/usr/include/freetype2 -I/usr/include/harfbuzz -I/usr/include/libdrm -I/usr/include/libpng16 -I/usr/include/gdk-pixbuf-2.0 -I/usr/include/glib-2.0 -I/usr/lib/glib-2.0/include -D_FORTIFY_SOURCE=2  -march=x86-64 -mtune=generic -O2 -pipe -fstack-protector-strong -fno-plt -U_FORTIFY_SOURCE -D_FORTIFY_SOURCE=1       
Linking: gcc   -L. -Wl,-O1,--sort-common,--as-needed,-z,relro -fstack-protector -rdynamic -Wl,-export-dynamic -Wl,-E -Wl,-rpath,/usr/lib/perl5/core_perl/CORE  -Wl,-O1,--sort-common,--as-needed,-z,relro,-z,now -L/usr/local/lib -Wl,--as-needed -o vim   -lgtk-3 -lgdk-3 -lpangocairo-1.0 -lpango-1.0 -latk-1.0 -lcairo-gobject -lcairo -lgdk_pixbuf-2.0 -lgio-2.0 -lgobject-2.0 -lglib-2.0 -lSM -lICE -lXt -lX11 -lXdmcp -lSM -lICE  -lm -lncurses -lelf -lnsl    -lacl -lattr -lgpm -ldl   -Wl,-E -Wl,-rpath,/usr/lib/perl5/core_perl/CORE -Wl,-O1,--sort-common,--as-needed,-z,relro -fstack-protector-strong -L/usr/local/lib  -L/usr/lib/perl5/core_perl/CORE -lperl -lpthread -lnsl -ldl -lm -lcrypt -lutil -lc   -L/usr/lib -ltclstub8.6 -ldl -lz -lpthread -lieee -lm     
```

I first encountered this bug while using `fish`, but after switching to `bash` I could still reproduce this behavior. So I don't the choice of shell is relevant. But I'll attach the info just in case:

```
 ~ ) fish --version
fish, version 2.6.0

 ~ ) bash --version
GNU bash, version 4.4.12(1)-release (x86_64-unknown-linux-gnu)
Copyright (C) 2016 Free Software Foundation, Inc.
License GPLv3+: GNU GPL version 3 or later <http://gnu.org/licenses/gpl.html>

This is free software; you are free to change and redistribute it.
There is NO WARRANTY, to the extent permitted by law.
```

**Comment by Tyriar** [comment]
You didn't add the "settle:settle:"? I don't see how vscode could have done thus.

**Comment by lfairy** [comment]
I tried to reproduce the issue today, and got the text "Foo" instead:

![screenshot from 2017-08-28 20-12-54](https://user-images.githubusercontent.com/1572323/29764945-559bd0c8-8c2d-11e7-86ac-73a63ffb14d0.png)

This was after I renamed my `~/.vimrc` and `~/.vim`, so it can't be a problem with my Vim config.

@Tyriar is there any other way I can help with debugging this issue?

**Comment by Tyriar** [comment]
@lfairy could you try setup xterm.js and run the demo? This will tell us if it's something in VS Code or the upstream library https://github.com/sourcelair/xterm.js#demo

**Comment by lfairy** [comment]
@Tyriar thanks for the suggestion. I tested with 1.1.3, 2.0.1, 2.9.2 and the tip of the v3 branch (38c66e7c1a65ed9e293fcca1a22f0b80c6686439), running `git clean -fdx` before each build, and reproduced the issue with all four versions. Using node 8.4.0 and npm 5.3.0 on Arch Linux.

![screenshot of xterm.js from 2017-08-30 22-30-08](https://user-images.githubusercontent.com/1572323/29868397-d491e506-8dd2-11e7-8c3e-688e81deb004.png)

I also tried to test 1.0.0 but that had a node-gyp dependency (`pty.js@0.3.0`) that failed to compile.
